File: _src/07/Chapter7.VB.Data/Maps/PersonMap.vb
Imports System.Data.Entity.ModelConfiguration
Imports Chapter7.VB.Data.Models

Namespace Maps

    Public Class PersonMap
        Inherits EntityTypeConfiguration(Of Person)

        Public Sub New
             Me.Property(Function(p) p.FirstName) _ 
                .HasMaxLength(30) _ 
                .IsRequired()
            Me.Property(Function(p) p.LastName) _ 
                .IsRequired() _ 
                .HasMaxLength(30)
            Me.Property(Function(p) p.NickName) _ 
                .HasMaxLength(40) _ 
                .IsRequired()
            Me.Property(Function(p) p.Age) _ 
                .IsRequired()
        End Sub
        
    End Class
End Namespace

File: _src/07/Chapter7.CSharp.Data/Maps/PersonMap.cs
using System.Data.Entity.ModelConfiguration;
using Chapter7.CSharp.Data.Models;

namespace Chapter7.CSharp.Data.Maps
{
    public class PersonMap: EntityTypeConfiguration<Person>
    {
        public PersonMap()
        {
            Property(p => p.FirstName)
                .HasMaxLength(30)
                .IsRequired();
            Property(p => p.LastName)
                .IsRequired()
                .HasMaxLength(30);
            Property(p => p.NickName)
                .HasMaxLength(40)
                .IsRequired();
            Property(p => p.Age)
                .IsRequired();
        }
    }
}
